tsconfig.json

```diff
@@ -0,0 +1,69 @@
+{
```

Comment:
I left all these options in w/ the provided explanations so that people could review them. Will cut down to only those in use before merging

Reply:
will defer to @drewj @ericpias @ondrosh here

Reply:
The enabled options seem reasonable to me when just adding typings.